ci/scenario_helpers/vm_handler.py

```diff
@@ -314,35 +313,43 @@ def listener(self, listening_host, port_queue, message_queue, result_queue, time
                 port = listening_socket.getsockname()[1]
                 port_queue.put(port)
                 listening_socket.listen(5)
-                while message_queue.empty():
-                    if time.time() - start_time > self.SOCKET_BINDING_TIMEOUT:
-                        raise RuntimeError('Trying to receive messages timed out after {0}s'.format(self.SOCKET_BINDING_TIMEOUT))
-
-                    time.sleep(0.5)
-                while not message_queue.empty():
-                    connection_messages = message_queue.get()
-                    self.LOGGER.debug('Connection message received: {0}'.format(connection_messages))
-                    try:
-                        while len(connection_messages) > 0:  # and not stop_event.is_set()
-                            if time.time() - start_time > timeout:
-                                raise RuntimeError('Listening timed out after {0}s'.format(timeout))
-                            conn, addr = listening_socket.accept()
-                            self.LOGGER.debug('Connected with {0}:{1}'.format(addr[0], addr[1]))
-                            data = conn.recv(1024)
-                            self.LOGGER.debug('Connector said {0}'.format(data))
-                            if data in connection_messages:
-                                connection_messages.remove(data)
-                                vm_name = data.rsplit('_', 1)[-1]
-                                self.LOGGER.debug('Recognized sender as {0}'.format(vm_name))
-                                vm_ips_info[vm_name] = {'ip': addr[0]}
-                        pass
-                    finally:
-                        message_queue.task_done()
-            except socket.error as ex:
-                self.LOGGER.error('Could not bind the socket. Got {0}'.format(str(ex)))
-                raise
+                try:
+                    conn, addr = listening_socket.accept()
```

Comment:
This will still be blocking all your threading when not all messages have come in